src/main/kotlin/com/github/tomasmilata/intelliroutes/RoutesGotoControllerHandler.kt

```diff
@@ -8,6 +8,7 @@ import com.intellij.psi.JavaPsiFacade
 import com.intellij.psi.PsiElement
 import com.intellij.psi.impl.source.tree.LeafPsiElement
 import com.intellij.psi.search.ProjectScope
+import org.jetbrains.plugins.scala.finder.ScalaClassFinder
```

Comment:
I'm afraid we won't be able to use `ScalaFinder` here because the scala plugin is only an optional dependency, so this would fail in runtime if the scala plugin is not present.

I'm thinking about 2 alternatives now:

- Implementing 2 separate `GotoContributionHandler`s, in a similar way to `JavaRoutesCompletionContributor` vs `ScalaRoutesCompletionContributor`, referenced from `plugin.xml` vs `plugin-scala.xml`
- Making the scala plugin dependency mandatory. I wonder what's the proportion of Play users using Java and not Scala and if Java users are willing to install Scala plugin even if they're not using other scala features.

Any thoughts?

Reply:
Thank you Tomas. I've split the goto handler into two. I have exactly zero knowledge of IDEA plugin development and Lex/Flex so it is pretty much just a copy of the existing logic :-P

Reply:
Haha, that's how we learn!

Thanks for splitting it.